### ISSUE TITLE
Inconsistent performance and CPU utilization

### ISSUE THREAD

**Issue by ldmtwo** [Microsoft/CNTK#147]
The compute structure of CNTK is very unclear. I get that GPUs are each one MPI process, but let's ignore GPUs altogether. When I ran the MNIST 01_OneHidden (below was renamed), I saw one process with 7200% CPU (Xeon E5v3 = 72VCores total). The output reported 1150-1250 samples/sec. 50 epochs took 41m16s. That was this command. Also, there are no GPUs, just the 2 socket Xeon above.

Within /home/hadoop/bin/CNTK/Examples/Image/MNIST/Data
`cntk configFile=/home/hadoop/bin/CNTK/Examples/Image/MNIST/Config/masterTest.cntk `

I then run it with MPI using just the master and NP=2.
`mpirun --wdir /home/hadoop/bin/CNTK/Examples/Image/MNIST/Data --np 2  cntk configFile=/home/hadoop/bin/CNTK/Examples/Image/MNIST/Config/masterTest.cntk parallelTrain=true`

I now observe 2 processes with 200% CPU (2x200 = 400%). I was expecting contention. 50 epochs ran at 1990-2020 samples/sec. It finished in 25m0s. 

3rd test: NP=16, using master+node02, 220-260 samples/sec, 900% CPU (8x900 = 7200% per node)
 
How do I explain this performance? 

Sample script:
```
rm -fr ../Output

ssh node02 "rm -fr /home/hadoop/bin/CNTK/Examples/Image/MNIST/Output"
ssh master "rm -fr /home/hadoop/bin/CNTK/Examples/Image/MNIST/Output"

mpirun --wdir /home/hadoop/bin/CNTK/Examples/Image/MNIST/Data --np 16 --host master,node02 cntk configFile=/home/hadoop/bin/CNTK/Examples/Image/MNIST/Config/masterTest.cntk parallelTrain=true
```

**Comment by amitaga** [comment]
Can you please share the full config you used in these runs?

**Comment by mradmila** [comment]
Same issue as #149

**Comment by duythongnguyen211** [comment]
I ran CNTK on my computer (Ubuntu 14.04, CPU Intel Core i5 vPro).
When I ran MNIST 01_OneHidden.cntk 30 epochs with 1 process, It took >4m, I saw one process with 300% CPU.
I ran it with 2 processes using "mpirun -np 2", I saw 2 processes with 100% CPU / process and it took 4m16s
Then I ran it with 8 processes using 2 CPUs (2 same machines) by the folowing command line:
"mpirun -np 8 -host master,slave1 cntk configFile=../Config/01_OneHidden.cntk parallelTrain=true"
i saw 4 processes with 70% CPU / process on master machine and 4 processes with 100% CPU / process on slave machine. It took 14m09s.
